### PR TITLE
tools: updated golangci-lint to v1.31.0

### LINKED ISSUE #71: golangci-lint installation fails
Running `make install-tools` fails with the following error:
```
GOBIN=/home/roeesh/the-mesh-for-data/./hack/tools/bin GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.19.1
go: finding github.com v1.19.1
go: finding github.com/golangci/golangci-lint/cmd/golangci-lint v1.19.1
go: finding github.com/golangci/golangci-lint/cmd v1.19.1
go: finding github.com/golangci v1.19.1
# github.com/golangci/golangci-lint/pkg/golinters
../go/pkg/mod/github.com/golangci/golangci-lint@v1.19.1/pkg/golinters/golint.go:50:14: l.LintASTFiles undefined (type *"github.com/golangci/lint-1".Linter has no field or method LintASTFiles)
make: *** [.mk/tools.mk:35: hack/tools/bin/golangci-lint] Error 2
```

Error is fixed when upgrading to golangci-lint v1.31.0

### PR DESCRIPTION
**Pull request by roee88**
This is required due to install failure of v1.19.1

Code style fixes identified by the new version are included

Fixes #71

### CODE REVIEW

**Comment by roee88** [comment]
@rohithdv please note that `log.Fatal()` is removed from two places in `test/services/pilot/catalog-credential-mock/mockup_pilot_catalog_con.go`  because it fails the lint check. The reason is that the `log.Fatal()` call disables the `defer conn.Close()` call.  Would be nice if you could improve the mocks code (remove commented out stuff, comment in go style, handle errors cleaner, etc.)

**Review by froesef — APPROVED** [review]
I had these errors as well in a different PR. Upgrading solved it for me. Agreeing on one version and fixing this in a smaller PR like this makes sense to me.

**Review by froesef — APPROVED** [review]

**Comment by roee88** [comment]
@rohithdv can you please take a loot?

File: test/services/pilot/policy-manager-mock/mockup_pilot.go [hunk 1]
```diff
@@ -82,13 +82,13 @@ func constructInputParameters() *pb.ApplicationContext {
 
 	// generalOperations := []*pb.AccessOperation{}
```
Comment (rohithdv) [inline test/services/pilot/policy-manager-mock/mockup_pilot.go:84]:
```suggestion

```

File: test/services/pilot/policy-manager-mock/mockup_pilot.go [hunk 2]
```diff
@@ -82,13 +82,13 @@ func constructInputParameters() *pb.ApplicationContext {
...
 
-	//applicationContext := &pb.ApplicationContext{PolicyManagerCredentials: policyManagerCredentials, AppInfo: applicationDetails, Datasets: datasets}
+	// applicationContext := &pb.ApplicationContext{PolicyManagerCredentials: policyManagerCredentials, AppInfo: applicationDetails, Datasets: datasets}
```
Comment (rohithdv) [inline test/services/pilot/policy-manager-mock/mockup_pilot.go:85]:
```suggestion
```

File: test/services/pilot/policy-manager-mock/mockup_pilot.go [hunk 3]
```diff
@@ -82,13 +82,13 @@ func constructInputParameters() *pb.ApplicationContext {
...
 	applicationContext := &pb.ApplicationContext{AppId: appID, AppInfo: applicationDetails, Datasets: datasets}
 
-	//ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
-	//ctx, cancel := context.WithTimeout(context.Background(), timeoutPeriod*time.Second)
+	// ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
+	// ctx, cancel := context.WithTimeout(context.Background(), timeoutPeriod*time.Second)
```
Comment (rohithdv) [inline test/services/pilot/policy-manager-mock/mockup_pilot.go:89]:
```suggestion
```

File: test/services/pilot/policy-manager-mock/mockup_pilot.go [hunk 4]
```diff
@@ -82,13 +82,13 @@ func constructInputParameters() *pb.ApplicationContext {
...
 
-	//defer cancel()
+	// defer cancel()
```
Comment (rohithdv) [inline test/services/pilot/policy-manager-mock/mockup_pilot.go:91]:
```suggestion	
```

File: test/services/pilot/catalog-credential-mock/mockup_pilot_catalog_con.go [hunk 5]
```diff
@@ -15,7 +15,6 @@ import (
 	pb "github.com/ibm/the-mesh-for-data/pkg/connectors/protobuf"
 
 	"google.golang.org/grpc"
```
Comment (rohithdv) [inline test/services/pilot/catalog-credential-mock/mockup_pilot_catalog_con.go:17]:
```suggestion
        "github.com/pkg/errors"
	"google.golang.org/grpc"
```

**Pull request merged**